spyder_env_manager/spyder/widgets/packages_table.py

```diff
@@ -0,0 +1,282 @@
+        if orientation == Qt.Horizontal:
+            if section == PACKAGE:
+                return to_qvariant(_("Package"))
+            elif section == DESCRIPTION:
+                return to_qvariant(_("Description"))
+            elif section == VERSION:
+                return to_qvariant(_("Version"))
+        return to_qvariant()
+
+    def rowCount(self, index=QModelIndex()):
+        """Qt Override."""
+        return len(self.packages)
+
+    def columnCount(self, index=QModelIndex()):
+        """Qt Override."""
+        return 3
+
+    def row(self, row_num):
+        """Get row based on model index. Needed for the custom proxy model."""
+        return self.packages[row_num]
+
+    def reset(self):
+        """ "Reset model to take into account new search letters."""
+        self.beginResetModel()
+        self.endResetModel()
+
+
+class EnvironmentPackagesTable(QTableView):
+
+    sig_update_package = Signal()
+    sig_uninstall_package = Signal()
+    sig_change_package_version = Signal()
+
+    def __init__(self, parent, text_color=None):
+        QTableView.__init__(self, parent)
+        self.menu = None
+        self.menu_actions = []
+        self.empty_ws_menu = None
+        self.update_action = None
+        self.uninstall_action = None
+        self.change_action = None
+        self._parent = parent
+        self.delete_queue = []
+        self.source_model = EnvironmentPackagesModel(self, text_color=text_color)
+        self.setModel(self.source_model)
+        self.setSelectionBehavior(QAbstractItemView.SelectRows)
+        self.setSelectionMode(QAbstractItemView.SingleSelection)
+        self.setSortingEnabled(True)
+        self.setEditTriggers(QAbstractItemView.AllEditTriggers)
+        self.verticalHeader().hide()
+        self.load_packages(False)
+
+    def contextMenuEvent(self, event):
+        """Setup context menu"""
+        row = self.rowAt(event.pos().y())
+        packages = self.source_model.packages
+        if not packages[row]["dependence"]:
+            self.update_action = create_action(
+                self, _("Update package(s)"), triggered=self.selection
+            )
+            self.uninstall_action = create_action(
+                self, _("Uninstall package(s)"), triggered=self.selection
+            )
+            self.change_action = create_action(
+                self,
+                _("Change package version with a version constraint"),
+                triggered=self.selection,
+            )
+            menu = QMenu(self)
+            self.menu_actions = [
+                self.update_action,
+                self.uninstall_action,
+                self.change_action,
+            ]
+            add_actions(menu, self.menu_actions)
+            menu.setMinimumWidth(100)
+            menu.popup(event.globalPos())
+            event.accept()
+
+    def focusOutEvent(self, e):
+        """Qt Override."""
+        super(EnvironmentPackagesTable, self).focusOutEvent(e)
```

Comment:
Since here only super is being call here, maybe overriding this method is not necessary?